### PR TITLE
Refactor mesh code from nif_import

### PR DESCRIPTION
**Pull request by neomonkeus**
@niftools/blender-nif-plugin-reviewer 

# Overview
Refactoring of the nif_import class to reduce lower level logic
- Remove mesh code from nif_import
- Reduce dependency chain
- Use base animation class and instantiate animation classes as required.

## Fixes Known Issues
[Ordered list of issues fixed by this PR]

## Documentation
Existing documentation

## Testing
Manually testing happy code paths, smoke test to ensure import worked

### Manual
Imported set of nifs to ensure that import worked as previously
- Single node mesh
- Grouped mesh
- Animation

### Automated
[List of tests run, updated or added to avoid future regressions]

## Additional Information
#313 should be merged before this as this PR is built on it and will make reviewing it easier
This should enable full implementation of property parsing 😸

### CODE REVIEW

File: io_scene_nif/modules/animation/animation_import.py [hunk 1]
```diff
@@ -149,7 +138,7 @@ def add_key(self, fcurves, t, key, interp):
         """
         Add a key (len=n) to a set of fcurves (len=n) at the given frame. Set the key's interpolation to interp.
         """
-        frame = round(t * self.fps)
+        frame = round(t * animation.FPS)
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/modules/animation/animation_import.py]: Not sure if this should be an class variable. Seems hacky idk. I would have all frames > seconds conversions use a function in a common scene or animation instance, one we make the anim modules inherit from.

File: io_scene_nif/modules/animation/material_import.py [hunk 2]
```diff
@@ -39,15 +39,13 @@
...
 from io_scene_nif.utility import nif_utils
 from io_scene_nif.utility.util_global import NifOp
 from io_scene_nif.utility.util_logging import NifLog
 
 
-class MaterialAnimation:
-
-    def __init__(self, parent):
-        self.animationhelper = parent
+class MaterialAnimation(Animation):
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/modules/animation/material_import.py]: Big kudos for inheritance! 👍 I had tried the same some time ago on my end, but ran into some cryptic errors that I was unable to debug lol

File: io_scene_nif/modules/geometry/mesh/mesh_import.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        # vertices
+        n_verts = n_tri_data.vertices
+
+        # vertex normals
+        n_norms = n_tri_data.normals
+
+        # v_map will store the vertex index mapping
+        # nif vertex i maps to blender vertex v_map[i]
+        v_map = [_ for _ in range(len(n_verts))]  # pre-allocate memory, for faster performance
+        # Following code avoids introducing unwanted cracks in UV seams:
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/modules/geometry/mesh/mesh_import.py]: todo: ditch all this code, optinally run remove doubles over output

File: io_scene_nif/modules/property/material/material_export.py [hunk 4]
```diff
@@ -59,8 +59,7 @@ def export_material_property(self, name, flags, ambient, diffuse, specular, emis
 
         # list which determines whether the material name is relevant or not  only for particular names this holds,
```
Comment (HENDRIX-ZT2) [inline io_scene_nif/modules/property/material/material_export.py]: todo: extract to function

File: io_scene_nif/modules/animation/animation_import.py (hunk 1 above)
Comment (neomonkeus) [reply io_scene_nif/modules/animation/animation_import.py]: Yeah I was calling it once as a static method of the global scene.
Main motivation was not having instance methods for now.

File: io_scene_nif/modules/animation/material_import.py (hunk 2 above)
Comment (neomonkeus) [reply io_scene_nif/modules/animation/material_import.py]: Yeah i reduced inatance method dependency ao was easier. You had done most of the leg work anyways

File: io_scene_nif/modules/property/material/material_export.py (hunk 4 above)
Comment (neomonkeus) [reply io_scene_nif/modules/property/material/material_export.py]: Yeah no idea why i formatted this. Yeah export code needs same TLC as import side

**Pull request merged**